Mathlib/Order/KrullDimension.lean

```diff
@@ -251,6 +251,73 @@ lemma height_le_coe_iff (x : α) (n : ℕ) :
   · simp
   · norm_cast
 
+/--
+The height of an element is infinite if there exist series of arbitrary length ending in that
+element.
+-/
+lemma height_eq_top_iff (x : α) :
+    height x = ⊤ ↔ ∀ n, ∃ p : LTSeries α, p.last = x ∧ p.length = n := by
+  constructor
+  · intro h n
+    apply exists_series_of_le_height x (n := n)
+    simp [h]
+  · intro h
+    rw [height_eq_iSup_last_eq, iSup_subtype', ENat.iSup_coe_eq_top, bddAbove_def]
+    push_neg
+    intro n
+    obtain ⟨p, hlast, hp⟩ := h (n+1)
+    exact ⟨p.length, ⟨⟨⟨p, hlast⟩, by simp [hp]⟩, by simp [hp]⟩⟩
+
+lemma height_eq_zero_iff (x : α) : height x = 0 ↔ (∀ y, ¬(y < x)) := by
```

Comment:
```suggestion
lemma height_eq_zero_iff {x : α} : height x = 0 ↔ IsMin x := by
```
and I would think that the `_iff` is inferrable
```suggestion
lemma height_eq_zero {x : α} : height x = 0 ↔ IsMin x := by
```

Reply:
There are lemmas like
```
lemma height_eq_zero_iff_isMin (a : α) :
    IsMin a ↔ height a = 0 := by
```
coming up next (https://github.com/leanprover-community/mathlib4/pull/15524/files#diff-f01f6dfbb94ae0c3a2d89d315b11b5fa88e9f864823c53d72130a3fe82adefb9R462). I guess then I can just delete this?

Reply:
Oh yes probably

Reply:
Included it here, as I need it below.